ipn/ipnlocal/local.go

```diff
@@ -2053,7 +2053,12 @@ func (b *LocalBackend) CheckIPNConnectionAllowed(ci *ipnauth.ConnIdentity) error
 		return errors.New("empty user uid in connection identity")
 	}
 	if uid != serverModeUid {
-		return fmt.Errorf("Tailscale running in server mode (uid=%v); connection from %q not allowed", serverModeUid, uid)
+		u, err := ipnauth.LookupUserFromID(b.logf, serverModeUid)
+		if err != nil {
+			return fmt.Errorf("Tailscale running in server mode (uid=%q); connection from %q not allowed", serverModeUid, uid)
+		} else {
+			return fmt.Errorf("Tailscale running in server mode (user=%q); connection from %q not allowed", u.Username, uid)
```

Comment:
these are still kinda gross with uids

Maybe make a little closure here that return a best effort stringification of a uid using ipnauth.LookupUserFromID and then use it in one fmt.Errorf return path for both the serverModeUid and uid?

Reply:
added a method to LocalBackend instead of a closure.